apps/hyperdrive-trading/src/ui/markets/YieldSourceCard/YieldSourceCard.tsx

```diff
@@ -1,58 +1,72 @@
-import { ArrowRightCircleIcon } from "@heroicons/react/16/solid";
 import {
   Protocol,
   findBaseToken,
+  findYieldSourceHyperdrives,
   findYieldSourceToken,
 } from "@hyperdrive/appconfig";
 import { ReactElement } from "react";
+import Skeleton from "react-loading-skeleton";
 import { useAppConfig } from "src/ui/appconfig/useAppConfig";
 import { Badge } from "src/ui/base/components/Badge";
 import { Well } from "src/ui/base/components/Well/Well";
+import { YieldSourceMarketsTable } from "src/ui/markets/YieldSourceMarketsTable/YieldSourceMarketsTable";
+import { useYieldSourceRate } from "src/ui/vaults/useYieldSourceRate";
 
 export function YieldSourceCard({
-  protocol,
+  yieldSourceProtocol,
 }: {
-  protocol: Protocol;
+  yieldSourceProtocol: Protocol;
 }): ReactElement {
   const appConfig = useAppConfig();
 
-  const pools = appConfig.hyperdrives.filter((hyperdrive) => {
-    const sharesToken = findYieldSourceToken({
-      tokens: appConfig.tokens,
-      yieldSourceTokenAddress: hyperdrive.sharesToken,
-    });
-    return sharesToken.extensions.protocol === protocol.id;
+  // Extract the list of pools that have this yield source
+  const pools = findYieldSourceHyperdrives({
+    yieldSourceId: yieldSourceProtocol.id,
+    hyperdrives: appConfig.hyperdrives,
+    tokens: appConfig.tokens,
   });
 
+  // The first pool is used to get the canonical base and shares tokens, etc..
+  const [firstPool] = pools;
+
   const {
     depositOptions: { isBaseTokenDepositEnabled, isShareTokenDepositsEnabled },
-  } = pools[0];
+  } = firstPool;
+
   const baseToken = findBaseToken({
-    baseTokenAddress: pools[0].baseToken,
+    baseTokenAddress: firstPool.baseToken,
     tokens: appConfig.tokens,
   });
   const sharesToken = findYieldSourceToken({
-    yieldSourceTokenAddress: pools[0].sharesToken,
+    yieldSourceTokenAddress: firstPool.sharesToken,
     tokens: appConfig.tokens,
   });
```

Comment:
Nit: Since we are destructuring already might make sense to grab the base and shares token here instead of using the dot syntax. Up to you though.
```suggestion
  const {
    depositOptions: { isBaseTokenDepositEnabled, isShareTokenDepositsEnabled },
    baseToken: baseTokenAddress,
    sharesToken: yieldSourceTokenAddress,
  } = firstPool;

  const baseToken = findBaseToken({
    baseTokenAddress,
    tokens: appConfig.tokens,
  });
  const sharesToken = findYieldSourceToken({
    yieldSourceTokenAddress,
    tokens: appConfig.tokens,
  });
```

Reply:
I actually ran into an issue with this that i'll address in the next PR. Basically if there are no pools in the appconfig we don't wanna assume there's a first pool to destructure from. This PR was getting a bit big so I left that edge case in for now.